packages/api/lambdas/sf-scheduler.js

```diff
@@ -3,64 +3,65 @@
 const get = require('lodash.get');
 const merge = require('lodash.merge');
 const uuidv4 = require('uuid/v4');
-const { getS3Object, parseS3Uri } = require('@cumulus/common/aws');
+const { getMessageFromTemplate } = require('@cumulus/ingest/queue');
 const { SQS } = require('@cumulus/ingest/aws');
 const { Provider, Collection } = require('../models');
 
-/**
- * Builds a cumulus-compatible message and adds it to the startSF queue
- * startSF queue will then start a stepfunction for the given message
- *
- * @param   {Object} event   - lambda input message
- * @param   {Object} context - lambda context
- * @param   {function} cb    - lambda callback
- */
-function schedule(event, context, cb) {
-  const template = get(event, 'template');
+const buildCumulusMeta = (queueName) => ({
+  execution_name: uuidv4(),
+  queueName
+});
+
+function getProvider(providerId) {
+  if (providerId) {
+    const p = new Provider();
+    return p.get({ id: providerId });
+  }
+  return {};
+}
+
+function getCollection(collection) {
+  if (collection) {
+    const c = new Collection();
+    return c.get({ name: collection.name, version: collection.version });
+  }
+  return {};
+}
+
+function buildMessage(event, messageTemplate) {
   const provider = get(event, 'provider', null);
   const meta = get(event, 'meta', {});
   const cumulusMeta = get(event, 'cumulus_meta', {});
   const collection = get(event, 'collection', null);
   const payload = get(event, 'payload', {});
-  let message;
+  const queueName = get(event, 'queueName', 'startSF');
+
+  messageTemplate.meta.provider = getProvider(provider);
+  messageTemplate.meta.collection = getCollection(collection);
+
+  return {
+    ...messageTemplate,
+    meta: merge(messageTemplate.meta, meta),
```

Comment:
A bit pedantic, but you may want to add the provider/collection by merging a third object here containing only those properties. Again, otherwise a `meta.provider` property would override it, which doesn't seem to be the intent of this code